python/paddle/fluid/executor.py

```diff
@@ -240,56 +241,63 @@ def run(self,
         if scope is None:
             scope = global_scope()
 
-        program = program.clone()
-        global_block = program.global_block()
+        program_cache_key = str(feed.keys() + fetch_list)
```

Comment:
Should the cache key includes a hash from the program as well? Otherwise if the program changed but the input / output is the same, we end up running a wrong program.

Reply:
yes, we are thinking about adding a version number to the program, so we can check if the program has changed.